spec/gruf/cli/executor_spec.rb

```diff
@@ -94,15 +94,47 @@
       end
     end
 
+    context 'when services are set via the CLI argument' do
+      let(:options_services) { [] }
+      let(:global_services) { [] }
+      let(:cli_services) { [Rpc::ThingService::Service] }
+      let(:args) { %w[--services Rpc::ThingService::Service] }
+
+      it 'adds each specified service to the server' do
+        expect(hook_executor).to receive(:call).with(:before_server_start, server: server).once
+        expect(server).to receive(:start!).once
+        cli_services.each do |svc|
+          expect(server).to receive(:add_service).with(svc).ordered
+        end
+        expect(logger).not_to receive(:fatal)
+        expect(hook_executor).to receive(:call).with(:after_server_stop, server: server).once
+        subject
+      end
+
+      context 'when the CLI argument has a service that does not exist' do
+        let(:args) { %w[--services FakeServiceThatIsNotReal::AtAll] }
+
+        it 'does not run the server' do
+          expect(server).not_to receive(:start!)
+          expect(hook_executor).not_to receive(:call).with(:before_server_start, server: server)
+          expect(hook_executor).not_to receive(:call).with(:after_server_stop, server: server)
+          expect { subject }.to raise_error(SystemExit)
```

Comment:
It would be nice if we could assert that the system exited with the human readable error message we expected for the given test (here and on L137).